pkgs/applications/editors/pinegrow/default.nix

```diff
@@ -0,0 +1,67 @@
+{ stdenv
+, lib
+, fetchurl
+, unzip
+, udev
+, nwjs
+, gcc-unwrapped
+, autoPatchelfHook
+, gsettings-desktop-schemas
+, gtk3
+, wrapGAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pinegrow";
+  version = "6.3";
+
+  src = fetchurl {
+    url = "https://download.pinegrow.com/PinegrowLinux64.${version}.zip";
+    sha256 = "0wldj633p67da077nfc67gr9xhq580rkfd0r3904sjq7x01r0kaz";
+  };
+
+  nativeBuildInputs = [
+    unzip
+    autoPatchelfHook
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    udev
+    nwjs
+    gcc-unwrapped
+    gsettings-desktop-schemas
+    gtk3
+  ];
+
+  sourceRoot = ".";
+
+  unpackPhase = "true";
+  # Extract and copy executable in $out/bin
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share/applications
+    mkdir -p $out/bin
+    unzip $src -d $out
+    substituteInPlace $out/Pinegrow.desktop \
+      --replace 'Exec=sh -c "$(dirname %k)/PinegrowLibrary"' 'Exec=sh -c "$out/bin/Pinegrow"'
+    mv $out/Pinegrow.desktop $out/share/applications/Pinegrow.desktop
+    ln -s $out/PinegrowLibrary $out/bin/Pinegrow
```

Comment:
`$out/PinegrowLibrary` should be moved to var/lib.

Reply:
just moving `PinegrowLibrary` results in a core dump due to missing dependencies in the same folder. 
One would have to move those to `$out/bin` as well. This clutters the bin directory unnecessary, which is why I chose a symlink.
I'm open to suggestions, otherwise I'd just leave it "as is".

Reply:
Could we move everything under lib and create a wrapper to cd/exec/etc. to fix this?

Reply:
I played a little bit more with it:

- The `/lib` folder already exists and is home to some libraries for pinegrow. WrapProgram successfully patches the ELF to link to those. So those aren't the problem. 
- `PinegrowLibrary` hardcodes the path to the other binaries, locales and resource files relative to its own location. This means I either move everything below the `PinegrowLibrary` in the directory structure in `$out/bin` or I symlink only the binary to `$out/bin`. I could obviously move everything under `$out/lib`as well and symlink the `$out/bin/PinegrowLibrary` to that location. I don't think that makes any sense, though. 
- Writing a wrapper script to cd to some other directory won't be possible, since the paths are hardcoded within the ELF binary itself. 

Did you mean something else by creating a wrapper? Because otherwise the simplest and cleanest solution would be, to just symlink the binary and to not clutter the `$out/bin` directory.

Reply:
> I could obviously move everything under $out/libas well and symlink the $out/bin/PinegrowLibrary to that location. I don't think that makes any sense, though.

It does make sense because we do not want to have ``$out/PinegrowLibrary` which translates to `/PinegrowLibrary`.

Reply:
ah, I see.
Unfortunately, Pinegrow starts with an empty page in the nw.js screen, when moved to `$out/lib` subfolder. Ironically, it works flawlessly when moved to an arbitrary directory != bin, != lib != libexec. 
For example:
```
unzip $src -d $out/test
substituteInPlace $out/test/Pinegrow.desktop \
      --replace 'Exec=sh -c "$(dirname %k)/PinegrowLibrary"' 'Exec=sh -c "$out/bin/Pinegrow"'
mv $out/test/Pinegrow.desktop $out/share/applications/Pinegrow.desktop
ln -s $out/test/PinegrowLibrary $out/bin/Pinegrow
runHook postInstall
```
works w/o problems. When I try the same with (e.g.) `$out/lib` or `$out/libexec` or a subfolder within, Pinegrow doesn't start up correctly.

Any idea, why? Or where I should move the files?

Reply:
I currently have no idea and not to much time to look much further into this. I think we should add a comment why it is done like it is done right now and why it can't be moved and then we should be good to go.

Reply:
ok, I just pushed an updated commit with your suggestion. If you are fine with the changes so far, I can squash the commits

Reply:
I think we want to move ``$out/pinegrow`` to something like ``$out/opt/pinegrow`` just to be FHS compliant.

Reply:
ok, done :+1: